paper-text-field-shared-styles.html

```diff
@@ -45,118 +61,72 @@
         outline: none;
       }
 
-      .floated-label-placeholder {
-        @apply(--paper-font-caption);
+      :host[no-float-label] #input {
+        padding: 0;
       }
 
-      /*****************************
-       * Underline
-       ******************************/
-      .underline {
-        position: relative;
-      }
+      label {
+        position: absolute;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        top: 16px;
 
-      .focused-line {
-        height: 2px;
+        pointer-events: none;
 
-        transform-origin: center center;
-        -webkit-transform-origin: center center;
-        transform: scale3d(0,1,1);
-        -webkit-transform: scale3d(0,1,1);
+        display: block;
+        width: 100%;
+        overflow: hidden;
+        white-space: nowrap;
+        text-align: left;
+        transition-duration: .2s;
+        transition-timing-function: cubic-bezier(.4,0,.2,1);
 
-        background: var(--paper-text-field-focus-color, --primary-color);
+        color: var(--paper-text-field-color, --secondary-text-color);
+        @apply(--paper-font-common-nowrap);
+        @apply(--paper-font-subhead);
 
-        @apply(--layout-fit);
+        transition-duration: .2s;
+        transition-timing-function: cubic-bezier(.4,0,.2,1);
       }
 
-      .underline.is-highlighted .focused-line {
-        -webkit-transform: none;
-        transform: none;
-        -webkit-transition: -webkit-transform 0.25s;
-        transition: transform 0.25s;
-
-        @apply(--paper-transition-easing);
+      :host([no-float-label]) label {
+        top: 8px;
       }
 
-      .unfocused-line {
-        height: 1px;
-        background: var(--paper-text-field-color, --secondary-text-color);
-
-        @apply(--layout-fit);
+      label.label-is-floating {
+        font-size: 12px;
+        top: 0px;
+        visibility: visible;
```

Comment:
I don't think the 'visibility` bit in `label.label-is-floating` is needed, though.